File: app/src/main/java/com/lkms/data/model/java/ProtocolItem.java
package com.lkms.data.model.java;

import com.google.gson.annotations.SerializedName;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Đại diện cho một hàng trong bảng "ProtocolItem".
 * Bảng này có khóa chính phức hợp (protocolId, itemId).
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class ProtocolItem {

    // Khóa ngoại đến Protocol
    @SerializedName("protocolId")
    private Integer protocolId;

    // Khóa ngoại đến Item
    @SerializedName("itemId")
    private Integer itemId;

    @SerializedName("quantity")
    private Integer quantity;
}

File: app/src/main/java/com/lkms/data/model/ProtocolItem.kt
// Trong file: data/model/ProtocolItem.kt
package com.lkms.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Đại diện cho một hàng trong bảng "ProtocolItem".
 * Bảng này có khóa chính phức hợp (protocolId, itemId).
 */
@Serializable
data class ProtocolItem(
    // Khóa ngoại đến Protocol
    @SerialName("protocolId")
    val protocolId: Int,

    // Khóa ngoại đến Item
    @SerialName("itemId")
    val itemId: Int,

    @SerialName("quantity")
    val quantity: Int? = null
)